packages/ui/__tests__/ui.spec.js

```diff
@@ -0,0 +1,270 @@
+import { Text } from '@instructure/ui-text'
+import { TextArea } from '@instructure/ui-text-area'
+import { TextInput } from '@instructure/ui-text-input'
+import { TimeSelect } from '@instructure/ui-time-select'
+import { ToggleDetails, ToggleGroup } from '@instructure/ui-toggle-details'
+import { Tooltip } from '@instructure/ui-tooltip'
+import { Tray } from '@instructure/ui-tray'
+import { TreeBrowser } from '@instructure/ui-tree-browser'
+import { TruncateText } from '@instructure/ui-truncate-text'
+import { View, ContextView } from '@instructure/ui-view'
+
+const EveryComponent = {
+  AccessibleContent,
+  PresentationContent,
+  ScreenReaderContent,
+  Alert,
+  Avatar,
+  Badge,
+  Billboard,
+  Breadcrumb,
+  BaseButton,
+  Button,
+  CloseButton,
+  CondensedButton,
+  IconButton,
+  ToggleButton,
+  Byline,
+  Checkbox,
+  CheckboxGroup,
+  CheckboxFacade,
+  ToggleFacade,
+  DateInput,
+  DateTimeInput,
+  Dialog,
+  DrawerLayout,
+  DrawerContent,
+  Editable,
+  InPlaceEdit,
+  Expandable,
+  FileDrop,
+  Flex,
+  FlexItem,
+  Focusable,
+  FormField,
+  FormFieldLabel,
+  FormFieldMessage,
+  FormFieldMessages,
+  FormFieldLayout,
+  FormFieldGroup,
+  Grid,
+  GridRow,
+  GridCol,
+  Heading,
+  Img,
+  Link,
+  List,
+  ListItem,
+  InlineList,
+  Menu,
+  MenuItem,
+  MenuItemGroup,
+  MenuItemSeparator,
+  MetricGroup,
+  Metric,
+  Modal,
+  Transition,
+  Navigation,
+  AppNav,
+  NumberInput,
+  Options,
+  Mask,
+  Overlay,
+  Pages,
+  Pagination,
+  Pill,
+  Popover,
+  Portal,
+  Position,
+  ProgressBar,
+  ProgressCircle,
+  RadioInput,
+  RadioInputGroup,
+  RangeInput,
+  Rating,
+  Responsive,
+  Select,
+  Selectable,
+  SimpleSelect,
+  Spinner,
+  InlineSVG,
+  SVGIcon,
+  Table,
+  Tabs,
+  Tag,
+  Text,
+  TextArea,
+  TextInput,
+  TimeSelect,
+  ToggleDetails,
+  ToggleGroup,
+  Tooltip,
+  Tray,
+  TreeBrowser,
+  TruncateText,
+  View,
+  ContextView
+}
+// some components need props in order to render
+const extraProps = {
+  TreeBrowser: {
+    collections: {
+      1: {
+        id: 1,
+        name: 'Assignments',
+        collections: [],
+        items: [1],
+        descriptor: 'Class Assignments'
+      }
+    },
+    items: {
+      1: { id: 1, name: 'Addition Worksheet' }
+    },
+    defaultExpanded: [1],
+    rootId: 1
+  },
+  InPlaceEdit: {
+    renderEditButton: jest.fn()
+  },
+  Navigation: {
+    label: 'Main navigation',
+    toggleLabel: {
+      expandedLabel: 'Minimize Navigation',
+      minimizedLabel: 'Expand Navigation'
+    }
+  },
+  Transition: {
+    children: undefined
+  }
+}
+
+describe('Testing every imported component for SSR', () => {
+  beforeAll(() => {
+    // since we don't provide every needed prop propTypes will complain when
+    // we render the components, so in order to not to spam the console
+    // we supress these warnings/errors
+    jest.spyOn(console, 'warn').mockImplementation(() => {})
+    jest.spyOn(console, 'error').mockImplementation(() => {})
```

Comment:
I think you need to call `mockRestore` after the test has run so it does not influence other tests https://jestjs.io/docs/mock-function-api#mockfnmockrestore

Reply:
Thanks, I changed the `restoreMocks` config to be true.